### PR TITLE
(Test) Add test cases for decrease_key method in Fibonacci heap #13

### PR DESCRIPTION
**Pull request by emmariaolsson**
**Contains**
- Test case 1: assertEqual for valid input and expected output
- Test case 2: assertTrue to verify output
- Test case 3: assertRaises for invalid key value as input (when Exception is raised)

### CODE REVIEW

**Comment by jvonhacht** [comment]
Ser bra ut. Måste dock också testa att sänka nyckeln på ett barn så att dens nyckel blir lägre än förälderns. Då måste trädet ändras och byggas om.

**Review by jonasjungaker — COMMENTED** [review]
Looks good and structured, might be good to add test cases to doc-string of the test function

**Comment by emmariaolsson** [comment]
Now I have added the test case that was requested as well as a negative test, resolve #13

File: tests/test_heap.py [hunk 1]
```diff
@@ -154,13 +162,19 @@ def test_fibonacci_heap_decrease_key(self):
             self.assertTrue(x.key in decrease_data)
 
         # test case 3: Negative
-        for x in fh._iterate(fh.root_list):
-            self.assertFalse(x.key in data) #OBS! Have to re-write to take into account [True True True False True]
+        nodes = self.get_all_nodes(fh, fh.root_list)
+        for i in nodes:
+            self.assertFalse(i.key in [3])
```
Comment (jvonhacht) [inline tests/test_heap.py]: `i.key == 3` is more readable than `i.key in [3]`

File: tests/test_heap.py [hunk 2]
```diff
@@ -154,13 +162,19 @@ def test_fibonacci_heap_decrease_key(self):
...
         #test case 4: Exception
         with self.assertRaises(Exception):
             fh.decrease_key(fh.find_min(), 8)
-    
+
+        #test case 5: After extract 
+        fh.extract_min_node()
+        fh.decrease_key(fh.root_list.child, 1)
+        self.assertEqual(fh.find_min().key, 1)
+        self.assertTrue(fh.find_min() in fh._iterate(fh.root_list))
```
Comment (jvonhacht) [inline tests/test_heap.py]: Should also check that `fh.root_list.child` is correct after the cuts. If root_list only had one child, this should be `None`, if it had multiple it should be `fh.root_list.child.right`.

File: tests/test_heap.py [hunk 3]
```diff
@@ -154,13 +162,19 @@ def test_fibonacci_heap_decrease_key(self):
...
```
Comment (JohanKJIP) [inline tests/test_heap.py]: Otherwise, everything looks great! 👍

**Review by jvonhacht — CHANGES_REQUESTED** [review]

**Comment by jvonhacht** [comment]
Otherwise, everything looks great! 👍

**Review by jvonhacht — APPROVED** [review]

**Comment by JohanKJIP** [comment]
Squash this merge

**Pull request merged**